### ISSUE TITLE
NoSemicolons rule false positives with enums

### ISSUE THREAD

**Issue by rocketraman** [detekt/detekt#6615]
## Expected Behavior

I have a class like this:

```
enum class Foo {
  FOO,
  BAR,
  ;

  fun foo() = Unit
}
```

This should not report any error from `NoSemicolons`, because the `;` here is necessary.

My detekt configuration I have

```
  NoSemicolons:
    active: true
    autoCorrect: true
```

## Observed Behavior

The `NoSemicolons` rule is triggered with detekt 1.23.3. It is *not* triggered with detekt 1.22.0.

## Your Environment

* Version of detekt used: 1.23.3
* Version of Gradle used (if applicable): 8.4
* Operating System and version: `Linux edison 6.5.9-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Oct 25 20:40:49 UTC 2023 x86_64 GNU/Linux`

**Comment by atulgpt** [comment]
Hi @rocketraman this rule is imported from Ktlint. Can you create the issue in Ktlint GitHub?

**Comment by rocketraman** [comment]
@atulgpt Apparently, the ktlint team cannot reproduce it. See https://github.com/pinterest/ktlint/issues/2351#issuecomment-1808976853.

**Comment by rocketraman** [comment]
As a side note, the fact that detekt integrates ktlint but does not take any responsibility for issues that arise due to that integration is a bit annoying. In that case, detekt should just de-integrate ktlint and avoid this problem in future. I commented about it here: https://github.com/detekt/detekt/discussions/5997#discussioncomment-7558857.

**Comment by 3flex** [comment]
> detekt integrates ktlint but does not take any responsibility for issues that arise due to that integration

This is not the case.

Detekt maintainers would normally suggest checking behaviour with ktlint directly (which is what the ktlint maintainer did after you raised the ticket there) first. That way we can confirm if it's more likely to be detekt, so we can track the issue here, or ktlint, where we'd close the issue here and ask that an issue be opened with ktlint project directly.

**Comment by marschwar** [comment]
Can you please provide a reproducer. I created a sample project and am not able to reproduce the issue with the code you provided in the issue description.
